app/scripts/controllers/backup.js

```diff
@@ -0,0 +1,75 @@
+import { exportAsFile } from '../../../shared/modules/export-utils';
+import { prependZero } from '../../../shared/modules/string-utils';
+
+export default class BackupController {
+  constructor(opts = {}) {
+    const {
+      preferencesController,
+      addressBookController,
+      trackMetaMetricsEvent,
+    } = opts;
+
+    this.preferencesController = preferencesController;
+    this.addressBookController = addressBookController;
+    this._trackMetaMetricsEvent = trackMetaMetricsEvent;
+  }
+
+  async restoreUserData(jsonString) {
+    const existingPreferences = this.preferencesController.store.getState();
+    const { preferences, addressBook } = JSON.parse(jsonString);
+    if (preferences) {
+      preferences.identities = existingPreferences.identities;
+      preferences.lostIdentities = existingPreferences.lostIdentities;
+      preferences.selectedAddress = existingPreferences.selectedAddress;
+
+      this.preferencesController.store.updateState(preferences);
+    }
+
+    if (addressBook) {
+      this.addressBookController.update(addressBook, true);
+    }
+
+    this._trackMetaMetricsEvent({
+      event: 'User restored data from backup',
```

Comment:
Has this event been scoped with product? 

Events should be title cased. 'User Restored Data From Backup', category likely needs to be 'Backup'.

Reply:
please use `User Data Imported`

```
 this._trackMetaMetricsEvent({
   event: 'User Data Imported',
```